spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -525,6 +527,18 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde {
    * @return
    *   The protobuf representation of the expression, or None if the expression is not supported
    */
+
+  def stringToEvalMode(evalModeStr: String): ExprOuterClass.EvalMode =
+    evalModeStr.toUpperCase(Locale.ROOT) match {
+      case "LEGACY" => ExprOuterClass.EvalMode.LEGACY
+      case "TRY" => ExprOuterClass.EvalMode.TRY
+      case "ANSI" => ExprOuterClass.EvalMode.ANSI
+      case _ =>
+        throw new IllegalArgumentException(
+          "Invalid eval mode"
```

Comment:
It would be good to include the invalid value in the error message

Reply:
I have updated the code as follows. 
```scala
   5   def stringToEvalMode(evalModeStr: String): ExprOuterClass.EvalMode =
   4     evalModeStr.toUpperCase(Locale.ROOT) match {
   3       case "LEGACY" => ExprOuterClass.EvalMode.LEGACY
   2       case "TRY" => ExprOuterClass.EvalMode.TRY
   1       case "ANSI" => ExprOuterClass.EvalMode.ANSI
 536       case invalid =>
   1         throw new IllegalArgumentException(
   2           s"Invalid eval mode '$invalid' "
   3         ) // Assuming we want to catch errors strictly
   4     }
```